src/BrowserInterop/Extensions/CallBackInteropWrapper.cs

```diff
@@ -18,9 +18,34 @@ public class CallBackInteropWrapper
 
         public bool GetJsObjectRef { get; set; }
 
+        /// <summary>
+        /// Adds javascript side debounce to the callback (in milliseconds)
+        /// see https://davidwalsh.name/javascript-debounce-function
+        /// </summary>
+        public double? Debounce { get; set; }
+
+        /// <summary>
+        /// Determines whether the callback should be invoked at the first occurrance 
+        /// or if the invocation shall be postponed until the debounce timeout is expired.
+        /// </summary>
+        public bool Immediate { get; set; }
+
+        /// <summary>
+        /// Determines whether the callback shall be invoked always when the debounce time is expired
+        /// or if it should be postponed until no event is incomming within one debounce time period.
+        /// </summary>
+        public bool TriggerPermanent { get; set; }
+
+        /// <summary>
+        /// Determines whether default values of the serialized callback data shall be transmitted
+        /// </summary>
+        public bool IncludeDefaults { get; set; }
```

Comment:
what is the point of this instead of passing "*"?

Reply:
lets say we have a classic person with name and address datastructure. Your current approach was to use the anonymous types only as serialization filters (e.g.):

`runtime.GetInstanceProperty<MyPerson>(objectRef,"person",new { FirstName = "*", Age="*", Address="*"})`

but I use the serialization filter as the data structure type itself

var data = await runtime.MyOwnGetInstanceProperty(objectRef,"person",new { FirstName = default(string) Age=default(int)})
Asssert.AreEqual(30,data.Age);

Therefore the string based serialization filtering is not suitable for me. I need a filter that is applied just by the anonymous properties and not by their value.

Reply:
```cs
runtime.GetInstanceProperty<MyPerson>(objectRef,"person","*")
```

will work too. The thing is, you can use "*" nearly everywhere. The serialization spec are here mostly for browser api where there can be loops or a lot of data (like window.document) that we don't want/need on .net side

Reply:
have a closer look at my sample. I don't specify any generic parameter. The type safetyness comes from the anonymous type. That can safe alot of boilerplate code on .NET side. I even considered to add some naming convention based access of sub object properties like: 
var data = await runtime.MyOwnGetInstanceProperty(objectRef,"person",new { Address__StreetNumber= default(int))
Asssert.AreEqual(10,data.Address__StreetNumber);

where the anonymous type property Address__StreetNumber is automatically translated to x['address']['streetNumber']

Reply:
What I already implemented too is typesafe filtering using expression trees. Means: I have a data structure, but i am able to filter in a typesafe manner for some of its properties e.g.:

```
var anonymousData = runtime.GetInstanceProperty<MyPerson>(objectRef,"person",x=> new {x.Name, x.Address.StreeNumber});
Asssert.AreEqual(10, anonyousData.StreetNumber);
```

Reply:
additional note: I just realized the whole processing of my examples could be done and cached on .NET side, by creating a Dictionary or JsonDocument serialization filter object per anonymous type and passing these filter objects then to your API. That means:  The change in js script would not be necessary. I'll consider that approach, because it might also reduce the serialization effort on .NET side if  you call a method with the same filter multiple times.